streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```diff
@@ -261,6 +261,16 @@ public TaskId taskId() {
         return config.originalsWithPrefix(prefix);
     }
 
+    @Override
+    public long currentSystemTimeMs() {
+        return timestamp;
```

Comment:
`timestamp` is the current record timestamp and thus should not be returned here (same below).

I seems, we need to actually extend this test-util class, what is a small change to the KIP. Note, that test-utils are public API and thus those changes also need a KIP. You don't need to worry about your KIP, we can just update it accordingly.

Seems, we need to have two more internal fields, one for each time. The public API change would be to add corresponding setters. We might also rename `setTimestamp` to `setRecordTimestamp` for clarity, we we would have three setters for record, system, and stream time.